kotlin/custom-dictionary-style/src/main/java/com/esri/arcgisruntime/sample/customdictionarystyle/MainActivity.kt

```diff
@@ -6,58 +6,83 @@ import com.esri.arcgisruntime.ArcGISRuntimeEnvironment
 import kotlinx.android.synthetic.main.activity_main.*
+import kotlinx.android.synthetic.main.style_controls_layout.*
 
 class MainActivity : AppCompatActivity() {
 
-  override fun onCreate(savedInstanceState: Bundle?) {
-    super.onCreate(savedInstanceState)
-    setContentView(R.layout.activity_main)
-
-    // authentication with an API key or named user is required to access basemaps and other 
-    // location services
-    ArcGISRuntimeEnvironment.setApiKey(BuildConfig.API_KEY)
-
-    // create a feature layer from a service feature table
-    val featureTable = ServiceFeatureTable(getString(R.string.restaurants_url))
-    val featureLayer = FeatureLayer(featureTable).apply {
-      // use a custom style to create a dictionary renderer and set it to the feature layer renderer
-      renderer = DictionaryRenderer(
-        DictionarySymbolStyle.createFromFile(
-          getExternalFilesDir(null)?.path + getString(R.string.restaurant_stylx_path)
-        )
-      )
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        // authentication with an API key or named user is required to access basemaps and other
+        // location services
+        ArcGISRuntimeEnvironment.setApiKey(BuildConfig.API_KEY)
+
+        // create a feature layer from a service feature table
+        val featureTable =
+            ServiceFeatureTable("https://services2.arcgis.com/ZQgQTuoyBrtmoGdP/arcgis/rest/services/Redlands_Restaurants/FeatureServer/0")
+        val featureLayer = FeatureLayer(featureTable)
+
+        // create a new map with a streets basemap and set it to the map view
+        mapView.map = ArcGISMap(BasemapStyle.ARCGIS_STREETS).apply {
+            // add the the feature layer to the map's operational layers
+            operationalLayers.add(featureLayer)
+            // set the initial viewpoint to the Esri Redlands campus
+            initialViewpoint = Viewpoint(34.0574, -117.1963, 5000.0)
+        }
+
+        // create a dictionary symbol style from the stylx file
+        val dictionarySymbolStyleFromFile =
+            DictionarySymbolStyle.createFromFile(getExternalFilesDir(null)?.path + "/Restaurant.stylx")
+        // create a new dictionary renderer from the dictionary symbol style
+        val dictionaryRendererFromFile = DictionaryRenderer(dictionarySymbolStyleFromFile)
+
+        // on style file click
+        styleFileRadioButton.setOnClickListener {
+            // set the feature layer renderer to the dictionary renderer from local stylx file
+            featureLayer.renderer = dictionaryRendererFromFile
+        }
+        // set the initial state to use the dictionary renderer from local stylx file
+        styleFileRadioButton.performClick()
+
+        // create a portal item using the portal and the item id of the dictionary web style
+        val portal = Portal("https://arcgisruntime.maps.arcgis.com")
```

Comment:
```suggestion
        val portal = Portal("https://arcgis.com")
```
Feel free to ignore this suggestion if it doesn't apply on Android.

On iOS, we have a convenient class method [`.arcGISOnline(withLoginRequired: false)`](https://developers.arcgis.com/ios/api-reference/interface_a_g_s_portal.html#a68398c03578deb91c307a96b8371a60d) so we typically don't use a string/URL literal. And generally we tend to cut off the prefix of the portal URL (i.e. the `arcgisruntime.maps.` portion).

Reply:
Oh cool. We don't have a convenience method like that!